modules/catalog-compaction/README.md

```diff
@@ -1 +1,101 @@
 # Catalog compaction module
+
+> **_NOTE_** Compaction has been moved to a separate module to eliminate circular dependencies,
+as it requires some components that may themselves depend on the catalog module. Please 
+refer to the catalog's module [readme](../catalog/README.md) to see more details about 
+catalog service and update log.
+
+## Overview
+
+During schema changes, the catalog update log stores incremental updates. Each update
+increases the catalog version. Over time, the log may grow to a humongous  size. To 
+address this, snapshotting was introduced to UpdateLog. Snapshotting means replacing 
+incremental updates with a snapshot.
+
+But different components can refer to a specific version of the catalog. Until they 
+finish their work with this version, it cannot be truncated.
+
+This module introduces [CatalogCompactionRunner](src/main/java/org/apache/ignite/internal/catalog/compaction/CatalogCompactionRunner.java)
+component. This component handles periodical catalog compaction, ensuring that dropped  versions
+of the catalog are no longer needed by any component in the cluster.
+
+## Compaction restrictions
+
+1. Catalog compaction can be performed up to the highest version (excluding) which activation time is less 
```

Comment:
What does excluding mean here? 
I also do not understand "which activation time..." part.

Reply:
For example, we have 3 versions of catalog

v1 activation time 1
v2 activation time 3
v3 activation time 5

and we have transaction with activation time 4

"excluding" means that we can trim only v1, because transaction uses catalog v2

Reply:
Catalog compaction can be performed up to the highest version, while excluding those with activation time lower...

Reply:
Thank you, I changed it as you suggested.
Sounds better and in my opinion the main meaning does not change.